### PR TITLE
fix(cli): block node command from exiting

### LINKED ISSUE #1510: Infinite loop after reaching `--debug.tip`
### Describe the bug

Once specified (with `--debug.tip`) block hash is reached, node starts rapidly switching between stages which looks like infinite loop because there are no more blocks to sync. 

I think ideally, sync should freeze after reaching the tip and node should still be responding for RPC calls etc.

I can work on a fix once we agree on the solution. 

### Steps to reproduce

```
# hash of block number 100
reth node --debug.tip 0xdfe2e70d6c116a541101cecbb256d7402d62125f6ddc9b607d49edc989825c64
```

### Node logs

```text
2023-02-22T14:34:56.671167Z  INFO Stage finished executing stage=SenderRecovery checkpoint=100
2023-02-22T14:34:56.671213Z  INFO Executing stage stage=Execution from=100
2023-02-22T14:34:56.671263Z  INFO Stage finished executing stage=Execution checkpoint=100
2023-02-22T14:34:56.671316Z  INFO Executing stage stage=MerkleUnwind from=100
2023-02-22T14:34:56.671319Z  INFO Stage finished executing stage=MerkleUnwind checkpoint=100
2023-02-22T14:34:56.671380Z  INFO Executing stage stage=AccountHashing from=100
2023-02-22T14:34:56.671385Z  INFO Stage finished executing stage=AccountHashing checkpoint=100
2023-02-22T14:34:56.671443Z  INFO Executing stage stage=StorageHashing from=100
2023-02-22T14:34:56.671448Z  INFO Stage finished executing stage=StorageHashing checkpoint=100
2023-02-22T14:34:56.671499Z  INFO Executing stage stage=MerkleExecute from=100
2023-02-22T14:34:56.671502Z  INFO Stage finished executing stage=MerkleExecute checkpoint=100
2023-02-22T14:34:56.671555Z  INFO Executing stage stage=TransactionLookup from=100
2023-02-22T14:34:56.671560Z  INFO Stage finished executing stage=TransactionLookup checkpoint=100
2023-02-22T14:34:56.671596Z  INFO Executing stage stage=IndexStorageHistory from=100
2023-02-22T14:34:56.671599Z  INFO Stage finished executing stage=IndexStorageHistory checkpoint=100
2023-02-22T14:34:56.671655Z  INFO Executing stage stage=IndexAccountHistory from=100
2023-02-22T14:34:56.671658Z  INFO Stage finished executing stage=IndexAccountHistory checkpoint=100
2023-02-22T14:34:56.671728Z  INFO Executing stage stage=Finish from=100
```

### Platform(s)

Mac (Apple Silicon)

### What version/commit are you on?

81fec4b375a1bdfc58a54c30aa15723336e5e98a

### Code of Conduct

- [X] I agree to follow the Code of Conduct

### PR DESCRIPTION
**Pull request by rkrasiuk**
Block the pipeline task after finishing.

Closes #1510 

This behavior will be changed with #1528

### CODE REVIEW

**Comment by rkrasiuk** [comment]
works as expected
Screenshot 2023-02-24 at 12 49 48

File: bin/reth/src/node/mod.rs [hunk 1]
```diff
@@ -193,8 +193,8 @@ impl Command {
 
         tx.await??;
 
-        info!(target: "reth::cli", "Finishing up");
-        Ok(())
+        info!(target: "reth::cli", "Pipeline has finished.");
```
Comment (mattsse) [inline bin/reth/src/node/mod.rs:196]: can we please add a note here why we're doing this in debug.tip mode

**Review by mattsse — CHANGES_REQUESTED** [review]

File: bin/reth/src/node/mod.rs (hunk 1 above)
Comment (rkrasiuk) [reply bin/reth/src/node/mod.rs:196]: yeah, sure

**Review by mattsse — APPROVED** [review]

**Comment by onbjerg** [comment]
You should just specify `--debug.max-block` to match `--debug.tip`. This is a fix for a user error IMO

**Comment by rkrasiuk** [comment]
@onbjerg well, to be fair, we instructed users to run `--debug.tip` from the beginning

**Comment by onbjerg** [comment]
To understand, we are already stopping execution after hitting `debug.tip`? Because I thought we discussed that and opted to not introduce that special logic (which is why we introduced `--debug.max-block` in the first place to leverage existing features)

**Comment by rkrasiuk** [comment]
@onbjerg that's correct, we do stop it, because i believe that's correct behavior. otherwise, the pipeline just loops around. can you point me to that discussion pls? i would've disagreed

**Comment by rkrasiuk** [comment]
@onbjerg regardless, this discussion is not relevant to this PR (rather to https://github.com/paradigmxyz/reth/pull/1522)

**Review by onbjerg — APPROVED** [review]

**Pull request merged**